frontend/components/music_player/player_hooks.js

```diff
@@ -0,0 +1,261 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+const Player = (props) => {
+  const [timeCounter, setTimeCounter] = useState(0);
+  const [prevVol, setPrevVol] = useState(100);
+  const [volume, setVolume] = useState(100);
+  const [volumeHover, setVolumeHover] = useState(false);
+  const [shuffle, setShuffle] = useState(false);
+  const [repeat, setRepeat] = useState(false);
+  const [songsPlayed, setSongsPlayed] = useState([]);
+  const [duration, setDuration] = useState([]);
+
+  let intervals = [];
+
+  const audioEl = React.createRef();
+  const progressSlider = React.createRef();
+  const volumeSlider = React.createRef();
+
+  // first useEffect is to load songs into the player's queue
+  // cleans up the player by setting universal state to 'pause' when player is closed
+  useEffect(() => {
+    props.getSongs().then((songs) => {
+      props.loadQueue(Object.values(songs.songs));
+    });
+    return () => {
+      props.pauseSong();
+    };
+  }, []);
+
+  // pause and play
+  const handleAction = () => {
+    if (props.songPlaying) {
+      props.pauseSong();
+      audioEl.current.pause();
+    } else {
+      props.playSong();
+      audioEl.current.play();
+    }
+  };
+
+  // if muting, saves the volume into a variable of 'previous volume', sets the volume to zero.
+  // if unmuting, sets the volume of the player to what the volume was before muting
+  const handleMute = () => {
```

Comment:
Similar to above, work towards a single-source of truth your props + state driving all other changes reactively.